cpp/src/io/parquet/reader_impl.cpp

```diff
@@ -281,7 +282,7 @@ void reader::impl::decode_page_data(size_t skip_rows, size_t num_rows)
         out_buf.user_data |= PARQUET_COLUMN_BUFFER_FLAG_LIST_TERMINATED;
       } else if (out_buf.type.id() == type_id::STRING) {
         // need to cap off the string offsets column
-        size_type const sz = static_cast<size_type>(col_sizes[idx]);
+        size_type const sz = static_cast<size_type>(col_string_sizes[idx]);
         cudaMemcpyAsync(static_cast<int32_t*>(out_buf.data()) + out_buf.size,
```

Comment:
This seems like a potential bug -- using `int32_t*` here and `sizeof(size_type)` below. Shouldn't these types match?
```suggestion
        cudaMemcpyAsync(static_cast<size_type*>(out_buf.data()) + out_buf.size,
```